spec/extensions/fast_fourier_transform_functions.md

```diff
@@ -0,0 +1,670 @@
+
+        - `'backward'`: No normalization.
+        - `'ortho'`: Normalize by `1/sqrt(n)`
+        - `'forward'`: Normalize by `1/n`.
+
+        Default: `'backward'`
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   An array transformed along the axes indicated by the `axes` keyword.
+
+#### Raises
+
+-   If `s` and `axes` have a different length.
+-   If an element of `axes` is larger than the number of axes of `a`.
+
+(function-fftshift)=
+### fftshift(x, /, *, axes=None)
+
+Reorders n-dimensional FTT data to have negative frequency terms first. In this way, the zero-frequency component shifts to the center of the spectrum. Note that `out[0]` is the Nyquist component only if the length of the input is even.
+
+#### Parameters
+
+-   **x**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **axes**: _Union\[ int, Sequence\[ int ], Tuple\[ int, ... ] ]_
+
+    -   Axes over which to shift. If not specified, it shifts all axes. Default: `None`.
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   The shifted array.
+
+(function-hfft)=
+### hfft(a, /, *, n=None, axis=-1, norm='backward')
+
+Computes the one-dimensional discrete Fourier Transform of a signal with Hermitian symmetry.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **n**: _int_
+
+    -   Length of the transformed axis of the output. If given, the input will be either zero-padded or trimmed to this length before computing the Hermitian FFT. Otherwise, it will default to `2 * (m - 1)` where `m` is the length of the input along the axis given by the `axis` keyword. Default: `None`.
+
+-   **axis**: _int_
+
+    -  Axis used to compute the fast fourier transform. If it is not specified, the last axis is used. Default: `-1`.
+
+-   **norm**: _str_
+
+    -   Specify the normalization mode. Should be one of the following modes:
+
+        - `'backward'`: No normalization.
+        - `'ortho'`: Normalize by `1/sqrt(n)`
+        - `'forward'`: Normalize by `1/n`.
+
+        Default: `'backward'`
+
+#### Returns
+
+-   **out**: _&lt;array&gt;_
+
+    -   A transformed array.
+
+#### Raises
+
+-   If `axis` is larger than the last axis of `a`.
+
+(function-ifft)=
+### ifft(a, /, *, n=None, axis=-1, norm='backward')
+
+Computes the one-dimensional inverse discrete Fourier Transform.
+
+#### Parameters
+
+-   **a**: _&lt;array&gt;_
+
+    -   Input array.
+
+-   **n**: _int_
+
+    -   Length of the transformed axis of the output. If given, the input will be either zero-padded or trimmed to the length `n` before computing the IFFT. Otherwise, the length of the input along the axis given by the `axis` keyword. Default: `None`.
+
+-   **axis**: _int_
+
+    -  Axis used to compute the fast fourier transform. If it is not specified, the last axis is used. Default: `-1`.
+
+-   **norm**: _str_
+
+    -   Specify the normalization mode. Should be one of the following modes:
+
+        - `'backward'`: No normalization.
```

Comment:
Is this correct? If the `backward` means no normalization in `fft`, then here in `ifft` it should be `1/n` right? See https://numpy.org/doc/stable/reference/routines.fft.html#normalization

Reply:
I took this ones from Pytorch docs, for example https://pytorch.org/docs/stable/fft.html#torch.fft.ifftshift

I'm not sure if they are doing different things or if they are the same. I'll investigate on this 🕵🏼‍♀️

Reply:
I found my mistake! I'm fixing this now (: